spateo/plotting/static/contour.py

```diff
@@ -0,0 +1,142 @@
+        swap_x_y: Whether or not to swap the x, y values when using `pl.imshow`.
+        background: string or None (optional, default 'None`)
+            The color of the background. Usually this will be either
+            'white' or 'black', but any color name will work. Ideally
+            one wants to match this appropriately to the colors being
+            used for points etc. This is one of the things that themes
+            handle for you. Note that if theme
+            is passed then this value will be overridden by the
+            corresponding option of the theme.
+        show: Visualize the result. Default to True.
+        save_show_or_return: `str` {'save', 'show', 'return', 'both', 'all'} (default: `show`)
+            Whether to save, show or return the figure. If "both", it will save and plot the figure at the same time. If
+            "all", the figure will be saved, displayed and the associated axis and other object will be return.
+        save_kwargs: `dict` (default: `{}`)
+            A dictionary that will passed to the save_fig function. By default it is an empty dictionary and the
+            save_fig function will use the {"path": None, "prefix": 'scatter', "dpi": None, "ext": 'pdf', "transparent":
+            True, "close": True, "verbose": True} as its parameters. Otherwise you can provide a dictionary that
+            properly modify those keys according to your needs.
+        **kwargs: Additional keyword arguments are all passed to :func:`imshow`.
+    """
+    import matplotlib.pyplot as plt
+    from matplotlib import rcParams
+    from matplotlib.colors import to_hex
+
+    label_list = np.unique(adata.obs[label_key])
+    labels = np.zeros(len(adata))
+    for i in range(len(label_list)):
+        labels[adata.obs[label_key] == label_list[i]] = i + 1
+
+    if bin_size is None:
+        bin_size = adata.uns["bin_size"]
+
+    label_img = np.zeros(
+        (
+            int(max(adata.obsm[spatial_key][:, 0] // bin_size)) + 1,
+            int(max(adata.obsm[spatial_key][:, 1] // bin_size)) + 1,
+        )
+    )
+    for i in range(len(adata)):
+        label_img[
+            int(adata.obsm[spatial_key][i, 0] // bin_size), int(adata.obsm[spatial_key][i, 1] // bin_size)
+        ] = labels[i]
+
+    contour_img = label_img.copy()
+    contour_img[:, :] = 255
+    for i in np.unique(label_img):
+        if i == 0:
+            continue
+        label_img_gray = np.where(label_img == i, 0, 1).astype("uint8")
+        _, thresh = cv2.threshold(label_img_gray, 0, 255, cv2.THRESH_BINARY_INV + cv2.THRESH_OTSU)
+        contour, _ = cv2.findContours(thresh, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_NONE)
+        contour_img = cv2.drawContours(contour_img, contour[:], -1, 0.5, 1)
+
+    fig, ax = plt.subplots(figsize=(5, 5), tight_layout=True)
+    kwargs.update({"cmap": "Blues"})
+
+    if swap_x_y:
+        im = ax.imshow(contour_img.T, **kwargs)
+    else:
+        im = ax.imshow(contour_img, **kwargs)
```

Comment:
Actually we need `contour_img` raw matrix to act as a background image in our anndata object. So there are two ways I can think of, ①add a function parameter to enable saving this image matrix to file, without adding any title or axis, so that the image (file) pixels match exactly with the matrix. Then we can load the image via `st.io.read_image`. ②enable returning this image matrix, and modify `st.io.read_image` to accept raw matrix input.

Reply:
looks the first one is good. I believe this is what Wang Shuai did when he add the background staining image?